### PR TITLE
Python 3.6 and 3.7

### PR DESCRIPTION
**Pull request by chrisrossi**
Bare minimum to get newer Pythons working. Python 3.4 was already failing tests before I got to it. Went ahead and dropped support. You're not likely to encounter 3.4 in the field, I don't think.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,8 +3,9 @@ sudo: false
 
 python:
   - "2.7"
-  - "3.4"
   - "3.5"
+  - "3.6"
+  - "3.7"
```
Comment (stevepiercy) [inline .travis.yml]: We might as well start testing on `3.8-dev` with allowed failures.

File: setup.py [hunk 2]
```diff
@@ -53,7 +54,6 @@
 testing_extras = [
     'nose',
     'coverage',
-    'mock',
```
Comment (stevepiercy) [inline setup.py]: Why was this moved from `testing_extras` into `install_requires`?

File: test_requirements.txt [hunk 3]
```diff
@@ -1,4 +1,3 @@
 coverage
-mock
```
Comment (stevepiercy) [inline test_requirements.txt]: Why was this removed?

File: tox.ini [hunk 4]
```diff
@@ -1,8 +1,9 @@
 [tox]
 envlist =
     py27,
-    py34,
     py35,
+    py36,
+    py37,
     cover,
     docs,
```
Comment (stevepiercy) [inline tox.ini]: Perhaps add `py38`, too?

**Comment by chrisrossi** [comment]
> Is mock required to run SubstanceD? 

Yes, it is.

See commit message for: https://github.com/Pylons/substanced/pull/305/commits/f3e187988136a8155b9b6a5153f2ca5f58bd466c

>  Also consider adding testing on 3.8.

You're welcome to do that.

**Comment by stevepiercy** [comment]
Thanks for the ref to the commit message.

I don't use SubstanceD, so I won't be adding 3.8 testing support.
